query/src/storages/stage/stage_source.rs

```diff
@@ -190,20 +193,44 @@ impl StageSource {
         let path = file_name;
         let object = op.object(&path);
 
+        // TODO(xuanwo): we need to unify with MultipartFormat.
+        let compression_algo = match stage.file_format_options.compression {
+            StageFileCompression::Auto => todo!("we will support auto in the future"),
+            StageFileCompression::Gzip => Some(CompressAlgorithm::Gzip),
+            StageFileCompression::Bz2 => Some(CompressAlgorithm::Bz2),
+            StageFileCompression::Brotli => Some(CompressAlgorithm::Brotli),
+            StageFileCompression::Zstd => Some(CompressAlgorithm::Zstd),
+            StageFileCompression::Deflate => Some(CompressAlgorithm::Deflate),
+            StageFileCompression::RawDeflate => todo!("we will support raw deflate in the future"),
+            StageFileCompression::Lzo => todo!("we will support lzo in the future"),
+            StageFileCompression::Snappy => todo!("we will support snappy in the future"),
+            StageFileCompression::None => None,
+        };
+        info!(
+            "Read stage {} file {} via compression {:?}",
+            stage.stage_name, &path, compression_algo
+        );
+
         // Get the format(CSV, Parquet) source stream.
         let source = match &file_format {
             StageFileFormatType::Csv => Ok(Self::csv_source(
                 ctx.clone(),
                 self.schema.clone(),
                 stage,
-                Box::new(object.reader().await?),
+                match compression_algo {
```

Comment:
Why not process the reader in previous ?
```suggestion
               let reader = match compression_algo {
                    None => Box::new(object.reader().await?),
                    Some(algo) => Box::new(object.decompress_reader_with(algo).await?),
                };
                
               let source =  match file_format ...
```

Reply:
Because `parquet` requires a `seekable_reader` here. The compression_algo match logic is duplicated in multipart format and stage, I will refactor them in the future.